Lib/email/utils.py

```diff
@@ -103,12 +104,127 @@ def formataddr(pair, charset='utf-8'):
     return address
 
 
+def _iter_escaped_chars(addr):
+    pos = 0
+    escape = False
+    for pos, ch in enumerate(addr):
+        if escape:
+            yield (pos, '\\' + ch)
+            escape = False
+        elif ch == '\\':
+            escape = True
+        else:
+            yield (pos, ch)
+    if escape:
+        yield (pos, '\\')
+
+
+def _strip_quoted_realnames(addr):
+    """Strip real names between quotes."""
+    if '"' not in addr:
+        # Fast path
+        return addr
+
+    start = None
+    remove = []
+    for pos, ch in _iter_escaped_chars(addr):
+        if ch == '"':
+            if start is None:
+                start = pos
+            else:
+                remove.append((start, pos))
+                start = None
+
+    result = []
+    pos = 0
+    for start, stop in remove:
+        result.append(addr[pos:start])
+        pos = stop + 1
+    result.append(addr[pos:])
+
+    return ''.join(result)
```

Comment:
nitpick: AFAIK this would be shorter and easier to follow with a `keep` list instead of `remove`.

Reply:
I rewrote the code to have a single loop.